src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -23,23 +20,40 @@ import {
   parseError,
 } from "@microsoft/vscode-azext-utils";
 import {
+  BicepDeploymentParametersResponse,
   BicepDeploymentScopeParams,
   BicepDeploymentScopeResponse,
-  BicepDeploymentWaitForCompletionParams,
   BicepDeploymentStartParams,
   BicepDeploymentStartResponse,
+  BicepDeploymentWaitForCompletionParams,
+  BicepUpdatedDeploymentParameter,
+  ParametersFileUpdateOption,
 } from "../language";
+import {
+  LanguageClient,
+  TextDocumentIdentifier,
+} from "vscode-languageclient/node";
 import { findOrCreateActiveBicepFile } from "./findOrCreateActiveBicepFile";
 
 export class DeployCommand implements Command {
-  private _none: IAzureQuickPickItem = {
+  private _none: IAzureQuickPickItem<string> = {
     label: localize("none", "$(circle-slash) None"),
-    data: undefined,
+    data: "",
   };
-  private _browse: IAzureQuickPickItem = {
+  private _browse: IAzureQuickPickItem<string> = {
     label: localize("browse", "$(file-directory) Browse..."),
+    data: "",
+  };
+  private _yes: IAzureQuickPickItem = {
```

Comment:
FYI in the future you can use DialogResponses.yes/no from @microsoft/vscode-azext-utils

Reply:
You can resolve